### PR TITLE
feat(helm): Add Extra Labels to Helm Chart for Enhanced Metadata Management

### PR DESCRIPTION
**Pull request by matazr**
## What this PR does / why we need it

This pull request enhances the Helm chart within the Kor project by adding extra labels to the metadata. This update improves the flexibility and organization of Kubernetes resource management.

### Changes Made
- **Modified the helper template**: Updated the `_helpers.tpl` file to include additional labels.
- **Updated `spec.template.metadata.labels`**: Ensured that the labels are correctly placed within the Helm chart's metadata.
  - Updated helper function `_helpers.tpl`:
    ```yaml
    apiVersion: v1
    kind: Pod
    metadata:
      labels:
        app.kubernetes.io/name: {{ include "kor.name" . }}
        helm.sh/chart: {{ include "kor.chart" . }}
        app.kubernetes.io/instance: {{ .Release.Name }}
        app.kubernetes.io/managed-by: {{ .Release.Service }}
        extraLabel1: value1
        extraLabel2: value2
    ```

### Benefits
- **Improved Metadata Management**: Additional labels provide better control and organization of Helm chart deployments.
- **Enhanced Flexibility**: Allows for more granular and customized resource management within Kubernetes environments.

## PR Checklist

- [ ] This PR adds K8s exceptions (false positives)
- [ ] This PR adds new code
- [ ] This PR includes tests for any new code

- **Files Affected**: `templates/_helpers.tpl ` `templates/deployment.yaml` `templates/cronjob.yaml` `values.yaml`

Please review the changes and provide feedback. If there are any questions or further modifications needed, feel free to let me know. Thank you for your time and consideration in reviewing this pull request.

### CODE REVIEW

File: charts/kor/values.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ prometheusExporter:
     type: ClusterIP
     port: 8080
 
-  serviceMonitor:
+    serviceMonitor:
```
Comment (kvanzuijlen) [inline charts/kor/values.yaml:62]:
```suggestion
  serviceMonitor:
```

File: charts/kor/values.yaml [hunk 2]
```diff
@@ -75,6 +75,9 @@ prometheusExporter:
     targetLabels: []
     metricRelabelings: []
 
+# Custom labels to add into metadata
+additionalLabels: {}
```
Comment (kvanzuijlen) [inline charts/kor/values.yaml:79]: I think commonLabels and customLabels are more commonly used

File: charts/kor/Chart.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.4
+version: 0.1.5
```
Comment (doronkg) [inline charts/kor/Chart.yaml:18]: Let's bump appVersion to 4.0.0

File: charts/kor/README.md [hunk 4]
```diff
@@ -1,13 +1,14 @@
 # kor
 
-![Version: 0.1.4](https://img.shields.io/badge/Version-0.1.4-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.3.4](https://img.shields.io/badge/AppVersion-0.3.4-informational?style=flat-square)
+![Version: 0.1.5](https://img.shields.io/badge/Version-0.1.5-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.3.4](https://img.shields.io/badge/AppVersion-0.3.4-informational?style=flat-square)
```
Comment (doronkg) [inline charts/kor/README.md:3]: Let's bump AppVersion to 4.0.0

File: charts/kor/Chart.yaml (hunk 3 above)
Comment (kvanzuijlen) [reply charts/kor/Chart.yaml:18]: 0.4.0 I think, right?

Comment (doronkg) [reply charts/kor/Chart.yaml:18]: Yes, correct

**Review by yonahd — APPROVED** [review]
Lgtm
Thanks!

**Comment by doronkg** [comment]
Hi @matazr, although the PR was merged, there're several disparities with README.

The `AppVersion` still points to v0.3.4, and many keys refer to `prometheusExporter.service` rather than `prometheusExporter.serviceMonitor`.

**Comment by matazr** [comment]
@doronkg thanks for your feedback.
I put [this](https://github.com/yonahd/kor/pull/280) PR
can you review as well?

**Pull request merged**